sdk/schemaregistry/azure-data-schemaregistry/src/main/java/com/azure/data/schemaregistry/AbstractDataDeserializer.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.schemaregistry.client.SchemaRegistryObject;
+import com.azure.data.schemaregistry.client.SchemaRegistryClient;
+import com.azure.data.schemaregistry.client.SchemaRegistryClientException;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Common implementation for all registry-based deserializers.
+ */
+public abstract class AbstractDataDeserializer extends AbstractDataSerDe {
+    private final ClientLogger logger = new ClientLogger(AbstractDataDeserializer.class);
+
+    protected final Map<String, ByteDecoder> byteDecoderMap = new ConcurrentHashMap<>();
```

Comment:
Was this made `protected` to sub-classes could interact with it? Generally we stay away from protected as it is part of the public API.

Reply:
Updated to be private.  All access to this map is controlled via loadByteDecoder()